### PR TITLE
feat: remove streetname v2

### PR DESCRIPTION
**Pull request by pgallik**

### CODE REVIEW

File: src/StreetNameRegistry/Municipality/Municipality_StreetName.cs [hunk 1]
```diff
@@ -159,6 +159,23 @@ public void RemoveStreetNameHomonymAdditions(PersistentLocalId persistentLocalId
                 .RemoveHomonymAdditions(languages, GuardUniqueActiveStreetNameNames);
         }
 
+        public void RemoveStreetName(PersistentLocalId persistentLocalId)
+        {
+            var streetName = StreetNames.FindByPersistentLocalId(persistentLocalId);
+
+            if (streetName is null)
```
Comment (emalfroy) [inline src/StreetNameRegistry/Municipality/Municipality_StreetName.cs]: duplicate, this is thrown in FindByPersistentLocalId

File: src/StreetNameRegistry/Municipality/StreetNameCommandHandlerModule.cs [hunk 2]
```diff
@@ -10,6 +10,7 @@ namespace StreetNameRegistry.Municipality
     using Be.Vlaanderen.Basisregisters.GrAr.Common.Pipes;
     using Be.Vlaanderen.Basisregisters.GrAr.Provenance;
     using Commands;
+    using Events;
```
Comment (emalfroy) [inline src/StreetNameRegistry/Municipality/StreetNameCommandHandlerModule.cs]: Unused using

File: src/StreetNameRegistry.Api.BackOffice/StreetNameController-Remove.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        [HttpPost("{persistentLocalId}/acties/verwijderen")]
+        [ProducesResponseType(StatusCodes.Status202Accepted)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status412PreconditionFailed)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status500InternalServerError)]
+        [SwaggerResponseHeader(StatusCodes.Status202Accepted, "location", "string", "De URL van het aangemaakte ticket.")]
+        [SwaggerResponseExample(StatusCodes.Status400BadRequest, typeof(BadRequestResponseExamples))]
+        [SwaggerResponseExample(StatusCodes.Status412PreconditionFailed, typeof(PreconditionFailedResponseExamples))]
+        [SwaggerResponseExample(StatusCodes.Status500InternalServerError, typeof(InternalServerErrorResponseExamples))]
+        [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme, Policy = PolicyNames.Adres.DecentraleBijwerker)]
```
Comment (jvandaal) [inline src/StreetNameRegistry.Api.BackOffice/StreetNameController-Remove.cs:42]: Also needs policy of interne bijwerker

File: test/StreetNameRegistry.Tests/ProjectionTests/StreetNameLatestItemProjectionsV2Tests.cs [hunk 4]
```diff
@@ -506,6 +506,57 @@ await Sut
...
+                    new Envelope<StreetNameWasApproved>(new Envelope(streetNameWasApproved, streetNameWasApprovedMetadata)),
+                    new Envelope<StreetNameWasRemovedV2>(new Envelope(streetNameWasRemovedV2, streetNameWasRetiredV2Metadata)))
+                .Then(async ct =>
+                {
+                    var expectedStreetName = (await ct.FindAsync<StreetNameDetailV2>(streetNameWasProposedV2.PersistentLocalId));
+                    expectedStreetName.Should().NotBeNull();
+                    expectedStreetName.MunicipalityId.Should().Be(streetNameWasProposedV2.MunicipalityId);
+                    expectedStreetName.NisCode.Should().Be(streetNameWasProposedV2.NisCode);
+                    expectedStreetName.PersistentLocalId.Should().Be(streetNameWasProposedV2.PersistentLocalId);
+                    expectedStreetName.Removed.Should().BeTrue();
```
Comment (jvandaal) [inline test/StreetNameRegistry.Tests/ProjectionTests/StreetNameLatestItemProjectionsV2Tests.cs:549]: I think it suffices that only the Removed property is checked?

File: src/StreetNameRegistry.Api.BackOffice/StreetNameController-Remove.cs (hunk 3 above)
Comment (jvandaal) [reply src/StreetNameRegistry.Api.BackOffice/StreetNameController-Remove.cs:42]: Integration tests are also missing.

**Review by jvandaal — APPROVED** [review]

**Pull request merged**